### PR TITLE
Bug #5022 - Clear button not hittable by voiceover

### PR DESCRIPTION
**Pull request by garvankeeley**

### CODE REVIEW

File: Client/Frontend/Browser/URLBarView.swift [hunk 1]
```diff
@@ -141,6 +141,9 @@ class URLBarView: UIView {
 
     fileprivate lazy var scrollToTopButton: UIButton = {
         let button = UIButton()
+        // This button interferes with accessibility of the URL bar as it partially overlays it, and keeps getting the VoiceOver focus instead of the URL bar.
+        // @TODO: figure out if there is an iOS standard way to do this that works with accessibility.
+        button.isAccessibilityElement = false
```
Comment (garvankeeley) [inline Client/Frontend/Browser/URLBarView.swift]: This fixes #4992 but it also relates to this bug as it interferes with VoiceOver selection of URL bar buttons.

File: Client/Frontend/Browser/URLBarView.swift [hunk 2]
```diff
@@ -790,25 +796,30 @@ class ToolbarTextField: AutocompleteTextField {
...
-                    button.setImage(tintedClearImage, for: [])
-                }
-            }
         }
     }
+
+    // The default button size is 19x19, make this larger
+    override func clearButtonRect(forBounds bounds: CGRect) -> CGRect {
+        let r = super.clearButtonRect(forBounds: bounds)
+        let grow = CGFloat(16)
```
Comment (justindarc) [inline Client/Frontend/Browser/URLBarView.swift]: nit: `let grow: CGFloat = 16`

File: Client/Frontend/Widgets/AutocompleteTextField.swift [hunk 3]
```diff
@@ -239,8 +239,8 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
             clearButton.snp.makeConstraints { make in
                 make.height.centerY.equalToSuperview()
                 make.width.equalTo(40)
-                // Without this offset, the button moves 5 pixels when switching from UILabel mode to UITextField mode
-                make.right.equalToSuperview().offset(5)
+                // Without this offset, the button moves when switching from UILabel mode to UITextField mode
+                make.right.equalToSuperview().offset(5.5)
```
Comment (justindarc) [inline Client/Frontend/Widgets/AutocompleteTextField.swift]: wha??

Comment (garvankeeley) [reply Client/Frontend/Widgets/AutocompleteTextField.swift]: I know. This is our crazy widget overlaying that we have where the first tap creates a special fully selected overlay widget, and then the second tap hides that and enters a normal UITextField in the edit state. In the first state, we 'fake' a clear text button, so it has to align with the native one. The native one changes slightly as iOS updates. That was the value I needed to get this aligned :facepalm:

**Pull request merged**